README.md

```diff
@@ -12,17 +12,25 @@ tsschecker is a powerful tool to check TSS signing status on combinations of var
 tsschecker is not only meant to be used to check firmware signing status, but also to explore Apple's TSS servers.<br/>
 By using all of its customization possibilities, you might discover a combination of devices and firmware versions that is getting signed but wasn't getting signed before. 
 
-# About nonces:
-## recommended generators for saving tickets:
-* `0xbd34a880be0b53f3` // default on the Electra, Chimera and Odyssey jailbreak apps.
-* `0x1111111111111111` // default on the unc0ver jailbreak app.
+# About Nonces:
+A [Nonce](https://wikipedia.org/wiki/Cryptographic_nonce) ("Number-used-ONCE") is a randomly generated value that is used to randomize apple's signed hash blobs.
 
-## Nonce Entangling (Apple A12/S4 and newer)
-Newer devices, such as the iPhone XR or the Apple Watch Series 4 (and any device newer) have nonce-entangling.
+it is created by the device with a nonce seed (generator) and then hashes that seed to create the nonce.<br/>On arm64e devices the nonce is also encrypted with the device's UID Key, see "Nonce Entangling" for more details.
```

Comment:
Not exactly. Yes UID is involved, however, there is a constant key encrypted by the UID, that encrypted key then is used to encrypt the 16 digit boot nonce. Then its hashed to become apnonce.

Reply:
interesting, didn't think there was another key hidden in the arm64e devices.
i figured UID and GID were the only ones.

is the constant key also fused into the AP during provisioning after the UID key had been fused-in?
i'll get this updated with the new info

Reply:
There's not another key, its just the UID. However nonce is non encrypted directly. There is a constant value(which we know) that gets encrypted by the UID. That encrypted value is used as the key that then encrypts nonce. We know the constant but thats pretty useless since we don't know UID. This is the one from a12 I don't know if its constant across all arm64e. 56 82 41 65 65 51 E0 CD  F5 6F F8 4C C1 1A 79 EF

Reply:
how did you get the constant value?

if it's easy enough (without JTAG, etc...), i could grab it from a production A14 (t8101) device.

Reply:
hey cryptic, what's your thoughts on my updated documentation in e9e1cea?

Reply:
I'm not 100% sure you have the bit's correct 128vs256 so might need to double check those.

Also a8 has collisions? I was not aware. Or is that wrong. Also worth mentioning that recovery collisions work with x-10.2, patched in 10.2.1

Reply:
Siguza wrote about it here: https://www.theiphonewiki.com/w/index.php?title=AES_Keys&type=revision&diff=109503&oldid=70014

found under Key 0x8A3, thats what i based the updated description on.

yeah, tihmstar reported that A8/A8X do sometimes collide nonces in DFU mode, and people seem to have some success with it as well. see here: https://www.reddit.com/r/iOSDowngrade/comments/ak3r8m/discussion_a7_a88x_device_user_save_your_blobs/

Reply:
Not always 0x8A3, there's a few entries it can be found under. dimentio loops until its found. I actually added a print statement to dimentio to dump the nonce entanglement key.
After copying the key from dimentio, you can paste into this script: https://github.com/Cryptiiiic/aes_nonce

Reply:
nvm 0x8A3 is constant

Reply:
i can't compile or use dimentio though, i don't have a mac. so Xcode is out of my reach.
what i have at my immediate disposal is Windows 7, Windows 10 and Ubuntu 22.04.

though it seems that you found that the key is constant anyway across arm64e devices (or at least for A12-A14).

anyway, is the description for Nonce entangling good? i do see AES-128-CBC in the aes_nonce script so siguza's description seems to be accurate.

Reply:
Either compile it on iPhone or use linux toolchains from theos wiki.

Reply:
Yes all is  good